drake/solvers/MultipleTreeProblem.m

```diff
@@ -0,0 +1,463 @@
+classdef MultipleTreeProblem
+  
+  properties
+    robot
+    endEffectorId
+    xStart
+    xGoal
+    endEffectorPoint
+    goalConstraints
+    additionalConstraints
+    qNom
+    mergingThreshold
+    capabilityMap
+    graspingHand
+    minDistance
+    trees
+    nTrees
+    startPoints
+    activeCollisionOptions
+    status
+    iterations
+  end
+  
+  properties (Constant)
+    EXPLORING = 1
+    GOAL_REACHED = 2
+    SUCCESS = 1
+    FAIL_ITERATION_LIMIT = 11
+    FAIL_NO_FINAL_POSE = 12
+  end
+  
+  methods
+    
+    function obj = MultipleTreeProblem(robot, endEffectorId,...
+        xStart, xGoal, xStartAddTrees, goalConstraints, additionalConstraints, qNom, varargin)
+      
+      warning('off','Drake:RigidBodyManipulator:ReplacedCylinder');
+      opt = struct('mergingthreshold', 0.2, 'capabilitymap', [], 'graspinghand', 'right',...
+        'mindistance', 0.005, 'activecollisionoptions', struct(), 'ikoptions', struct(),...
+        'steerfactor', 0.1, 'orientationweight', 1, 'maxedgelength', 0.05,...
+        'angletol', 10*pi/180, 'positiontol', 1e-3, 'endeffectorpoint', [0; 0; 0]);
+      optNames = fieldnames(opt);
+      nArgs = length(varargin);
+      if round(nArgs/2)~=nArgs/2
+        error('Needs propertyName/propertyValue pairs')
+      end
+      for pair = reshape(varargin,2,[])
+        inpName = lower(pair{1});
+        if any(strcmp(inpName,optNames))
+          opt.(inpName) = pair{2};
+        else
+          error('%s is not a recognized parameter name',inpName)
+        end
+      end
+      
+      obj.robot = robot;
+      obj.endEffectorId = endEffectorId;
+      obj.xStart = xStart;
+      obj.xGoal = xGoal;
+      obj.nTrees = size(xStartAddTrees, 2) + 2;
+      obj.goalConstraints = goalConstraints;
+      obj.additionalConstraints = additionalConstraints;
+      obj.qNom = qNom;
+      obj.mergingThreshold = opt.mergingthreshold;
+      obj.capabilityMap = opt.capabilitymap;
+      obj.graspingHand = opt.graspinghand;
+      obj.minDistance = opt.mindistance;
+      obj.activeCollisionOptions = opt.activecollisionoptions;
+      obj.endEffectorPoint = opt.endeffectorpoint;
+      obj.status = obj.EXPLORING;
+      
+      %compute sampling space
+      xyzMin = [min([xStart(1:2), xGoal(1:2)], [], 2) - 0.5; 0];
+      xyzMax = [max([xStart(1:2), xGoal(1:2)], [], 2) + 0.5; max([xStart(3), xGoal(3)]) + 1];
```

Comment:
These seem too problem specific. Could we have a `setTranslationSamplingBounds` method that sets the bounds for all of the trees?